Realm/RLMRealm.mm

```diff
@@ -183,35 +183,72 @@ + (instancetype)realmWithURL:(NSURL *)fileURL {
     return [RLMRealm realmWithConfiguration:configuration error:nil];
 }
 
+// The server doesn't send us the subscriptions for permission types until the
+// first subscription is created. This is fine for synchronous opens (if we're
+// creating a new Realm we create the permission objects ourselves), but it
+// causes issues for asyncOpen because it means that when our download completes
+// we don't actually have the full Realm state yet.
+static void waitForPartialSyncSubscriptions(Realm::Config const& config) {
+    auto realm = Realm::get_shared_realm(config);
+    auto table = ObjectStore::table_for_object_type(realm->read_group(), "__ResultSets");
+
+    realm->begin_transaction();
+    realm::sync::create_object(realm->read_group(), *table);
```

Comment:
What do you think about constructing this subscription object with a timeToLive which is several minutes in the future? Then it should clean itself up instead of having the server try to always evaluate it and fail which will add some small performance cost to the server.

Reply:
Or remove it manually when the notification completes?

Reply:
That's a good idea. Since the TTL is checked by the client when creating new subscriptions, I think it can actually just set expires_at to 0 and this object will always get cleaned up the first time the user creates a subscription.

It's theoretically possible for the ttl columns to not exist if it's an old reference realm that's never had a recent client connect to it before and the objectstore code wouldn't like if we create them here, but I guess it should be fine to just not set the expiration in that edge case.